### PR TITLE
Add documentation for hardware-accelerated compression codecs.

### PR DESCRIPTION
**Pull request by mulugetam**
### Description
Adds documentation for [hardware-accelerated compression codecs](https://github.com/opensearch-project/custom-codecs/pull/122) targeting OpenSearch 2.14. 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _im-plugin/index-codecs.md [hunk 1]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
 When an index segment is created, it uses the current index codec for compression. If you update the index codec, any segment created after the update will use the new compression algorithm. For specific operation considerations, see [Index codec considerations for index operations](#index-codec-considerations-for-index-operations).
 {: .note}
 
+As of OpenSearch 2.14, hardware-accelerated compression codecs for DEFLATE and LZ4 compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use their corresponding software implementations.
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:35]:
```suggestion
As of OpenSearch 2.14, hardware-accelerated compression codecs for the `DEFLATE` and `LZ4` compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use that platform's corresponding software implementations. 
```

File: _im-plugin/index-codecs.md [hunk 2]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
 When an index segment is created, it uses the current index codec for compression. If you update the index codec, any segment created after the update will use the new compression algorithm. For specific operation considerations, see [Index codec considerations for index operations](#index-codec-considerations-for-index-operations).
 {: .note}
 
+As of OpenSearch 2.14, hardware-accelerated compression codecs for DEFLATE and LZ4 compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use their corresponding software implementations. 
+
+The new hard-accelerated codecs can be used by setting the value of `index.codec`:
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:37]:
```suggestion
The new hardware-accelerated codecs can be used by setting one of the following values of the `index.codec` setting:
```

File: _im-plugin/index-codecs.md [hunk 3]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+* `qat_lz4` (OpenSearch 2.14 and later) -- uses hardware-accelerated LZ4.
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:38]:
```suggestion
* `qat_lz4` (OpenSearch 2.14 and later): Hardware-accelerated `LZ4`. 
```

File: _im-plugin/index-codecs.md [hunk 4]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
 When an index segment is created, it uses the current index codec for compression. If you update the index codec, any segment created after the update will use the new compression algorithm. For specific operation considerations, see [Index codec considerations for index operations](#index-codec-considerations-for-index-operations).
 {: .note}
 
+As of OpenSearch 2.14, hardware-accelerated compression codecs for DEFLATE and LZ4 compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use their corresponding software implementations. 
+
+The new hard-accelerated codecs can be used by setting the value of `index.codec`:
+* `qat_lz4` (OpenSearch 2.14 and later) -- uses hardware-accelerated LZ4. 
+* `qat_deflate` (OpenSearch 2.14 and later)  -- uses hardware-accelerated DEFLATE.
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:39]:
```suggestion
* `qat_deflate` (OpenSearch 2.14 and later): Hardware-accelerated `DEFLATE`. 
```

File: _im-plugin/index-codecs.md [hunk 5]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+As of OpenSearch 2.14, hardware-accelerated compression codecs for DEFLATE and LZ4 compression algorithms are available. These hardware-accelerated codecs are available on the latest 4th and 5th Gen Intel®️ Xeon®️ processors running Linux kernel 3.10 and later. For all other systems and platforms, the codecs use their corresponding software implementations. 
+
+The new hard-accelerated codecs can be used by setting the value of `index.codec`:
+* `qat_lz4` (OpenSearch 2.14 and later) -- uses hardware-accelerated LZ4. 
+* `qat_deflate` (OpenSearch 2.14 and later)  -- uses hardware-accelerated DEFLATE. 
+
+`qat_deflate` offers a much better compression ratio than `qat_lz4` with a modest drop in compression and decompression speed.
+{: .note}
+
+The `index.codec.compression_level` can be used to specify the compression level for both `qat_lz4` and `qat_deflate`. The new setting, `index.codec.qatmode`, controls the behavior of the hardware accelerator. It takes two valid values:
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:44]:
```suggestion
The `index.codec.compression_level` setting can be used to specify the compression level for both `qat_lz4` and `qat_deflate`. 

The `index.codec.qatmode` setting controls the behavior of the hardware accelerator and uses on of the following values:
```

File: _im-plugin/index-codecs.md [hunk 6]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+
+* `auto` -- uses hardware-accelerated compression. Fails over to use software if the hardware accelerator is not available.
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:46]:
```suggestion
* `auto`: If hardware acceleration fails, the algorithm switches to software acceleration.
```

File: _im-plugin/index-codecs.md [hunk 7]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+* `hardware` -- use hardware-accelerated compression. An exception is thrown if hardware is not available (guarantees a hardware-only compression).
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:47]:
```suggestion
* `hardware`: Guarantees hardware-only compression. If hardware is not available, an exception occurs until hardware exists.
```

File: _im-plugin/index-codecs.md [hunk 8]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+
+Refer to the snapshot section of this page for the implications `index.codec.qatmode` on snapshot and restore.
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:49]:
```suggestion
For information about the `index.codec.qatmode` settings implications on snapshots, see the [Snapshots](#snapshots) section.
```

File: _im-plugin/index-codecs.md [hunk 9]
```diff
@@ -32,6 +32,24 @@ For the `zstd` and `zstd_no_dict` codecs, you can optionally specify a compressi
...
+
+The hardware accelerator that is used is [Intel (R) QAT accelerator](https://www.intel.com/content/www/us/en/architecture-and-technology/intel-quick-assist-technology-overview.html).
```
Comment (Naarcha-AWS) [inline _im-plugin/index-codecs.md:51]:
```suggestion
For more information about hardware acceleration on Intel, see the [Intel (R) QAT accelerator overview](https://www.intel.com/content/www/us/en/architecture-and-technology/intel-quick-assist-technology-overview.html).
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 10]
```diff
@@ -79,10 +79,21 @@ OpenSearch supports the following static index-level index settings:
...
+- `index.codec.compression_level` (Integer): The compression level setting provides a trade-off between compression ratio and speed. A higher compression level results in a higher compression ratio (smaller storage size) with a trad-eoff in speed (slower compression and decompression speeds lead to greater indexing and search latencies). Can only be specified if index.codec is set to `zstd` and `zstd_no_dict` in OpenSearch 2.9 and later, and `qat_lz4` and `qat_deflate` in OpenSearch 2.14 and later. Valid values are integers in the [1, 6] range. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is 3.
+
+- `index.codec.qatmode` (String): The hardware acceleration mode used for `qat_lz4` and `qat_deflate` compression codecs:
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/index-settings.md:89]: Lets put this on one line like the other options in the list.

```suggestion
- `index.codec.qatmode` (String): The hardware acceleration mode used for the `qat_lz4` and `qat_deflate` compression codecs. Valid values are `auto` and `hardware`. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is `auto`. 
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 11]
```diff
@@ -79,10 +79,21 @@ OpenSearch supports the following static index-level index settings:
...
+- `index.codec.compression_level` (Integer): The compression level setting provides a trade-off between compression ratio and speed. A higher compression level results in a higher compression ratio (smaller storage size) with a trad-eoff in speed (slower compression and decompression speeds lead to greater indexing and search latencies). Can only be specified if index.codec is set to `zstd` and `zstd_no_dict` in OpenSearch 2.9 and later, and `qat_lz4` and `qat_deflate` in OpenSearch 2.14 and later. Valid values are integers in the [1, 6] range. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is 3.
+
+- `index.codec.qatmode` (String): The hardware acceleration mode used for `qat_lz4` and `qat_deflate` compression codecs: 
+
+    Valid values are:
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/index-settings.md:91]:
```suggestion

```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 12]
```diff
@@ -79,10 +79,21 @@ OpenSearch supports the following static index-level index settings:
...
+- `index.codec.compression_level` (Integer): The compression level setting provides a trade-off between compression ratio and speed. A higher compression level results in a higher compression ratio (smaller storage size) with a trad-eoff in speed (slower compression and decompression speeds lead to greater indexing and search latencies). Can only be specified if index.codec is set to `zstd` and `zstd_no_dict` in OpenSearch 2.9 and later, and `qat_lz4` and `qat_deflate` in OpenSearch 2.14 and later. Valid values are integers in the [1, 6] range. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is 3.
+
+- `index.codec.qatmode` (String): The hardware acceleration mode used for `qat_lz4` and `qat_deflate` compression codecs: 
+
+    Valid values are: 
+        
+    - `auto`
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/index-settings.md:93]:
```suggestion

```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 13]
```diff
@@ -79,10 +79,21 @@ OpenSearch supports the following static index-level index settings:
...
+    - `hardware`
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/index-settings.md:94]:
```suggestion

```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 14]
```diff
@@ -79,10 +79,21 @@ OpenSearch supports the following static index-level index settings:
...
+
+    Optional. Default is `auto`. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/).
```
Comment (Naarcha-AWS) [inline _install-and-configure/configuring-opensearch/index-settings.md:96]:
```suggestion

```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 15]
```diff
@@ -79,10 +79,17 @@ OpenSearch supports the following static index-level index settings:
     - `best_compression`
     - `zstd` (OpenSearch 2.9 and later)
     - `zstd_no_dict`(OpenSearch 2.9 and later)
+    - `qat_lz4`(OpenSearch 2.14 and later, on supported systems)
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:82]:
```suggestion
    - `qat_lz4` (OpenSearch 2.14 and later, on supported systems)
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 16]
```diff
@@ -79,10 +79,17 @@ OpenSearch supports the following static index-level index settings:
...
+    - `qat_deflate`(OpenSearch 2.14 and later, on supported systems)
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:83]:
```suggestion
    - `qat_deflate` (OpenSearch 2.14 and later, on supported systems)
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 17]
```diff
@@ -79,10 +79,17 @@ OpenSearch supports the following static index-level index settings:
...
+        
+    For `zstd`, `zstd_no_dict`, `qat_lz4`, and `qat_deflate` you can specify the compression level in the `index.codec.compression_level` setting. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is `default`.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:85]:
```suggestion
For `zstd`, `zstd_no_dict`, `qat_lz4`, and `qat_deflate`, you can specify the compression level in the `index.codec.compression_level` setting. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is `default`.
```

File: _install-and-configure/configuring-opensearch/index-settings.md [hunk 18]
```diff
@@ -79,10 +79,17 @@ OpenSearch supports the following static index-level index settings:
...
+
+- `index.codec.compression_level` (Integer): The compression level setting provides a trade-off between compression ratio and speed. A higher compression level results in a higher compression ratio (smaller storage size) with a trad-eoff in speed (slower compression and decompression speeds lead to greater indexing and search latencies). Can only be specified if index.codec is set to `zstd` and `zstd_no_dict` in OpenSearch 2.9 and later, and `qat_lz4` and `qat_deflate` in OpenSearch 2.14 and later. Valid values are integers in the [1, 6] range. For more information, see [Index codec settings]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/). Optional. Default is 3.
```
Comment (natebower) [inline _install-and-configure/configuring-opensearch/index-settings.md:87]: Please confirm that my changes retain the intended meaning.

**Pull request merged**